lib/chef/provider/package.rb

```diff
@@ -220,7 +220,7 @@ def removing_package?
       end
 
       def action_lock
-        if package_locked(new_resource.name, new_resource.version) == false
+        if package_locked(new_resource.package_name, new_resource.version) == false
```

Comment:
can you make this `unless package_locked` and just `if package_locked` below?  its not 100% logically equivalent but the case of not-true/not-false values like nil would be false in both cases, which is pretty buggy and incoherent anyway i think?

you could introduce:

```ruby
def package_locked?
  !!package_locked
end
```

to clean up the API a bit (subclasses in external code outside of core chef might be using `package_locked` without the `?` mark so I don't think you can change that there, its somewhat implicitly a public API and changing it for a nitpick is probably not worth it).

Reply:
I like the change to just `unless` and `if`. 

Unsure if I'm just not following the ramifcations of a `package_locked?` method in regards to cleaning up the code, but I'm missing something there as I don't understand how it would actually clean up the API.

Reply:
that would let you `if package_locked?` in the superclass where it would still call `package_locked` in the subclass but would be coerced to a boolean by `!!` and make the superclass at least read a little more idiomatically ruby.  IDK tho.  that was more of a thought.